website/registrations/views.py

```diff
@@ -289,7 +291,10 @@ def get_form(self, form_class=None):
 
     def post(self, request, *args, **kwargs):
         request.POST = request.POST.dict()
-        if request.member.latest_membership.type == Membership.BENEFACTOR:
+        if (
+            request.member.latest_membership.type == Membership.BENEFACTOR
+            or request.member.latest_membership.study_long
```

Comment:
If I'm not mistaken, the frontend will stull allow selecting 'member' instead of benefactor? You can prevent that by customizing the field's 'choices' as is done a few lines above this

Reply:
It will not